core/amber/src/main/scala/edu/uci/ics/texera/workflow/operators/download/BulkDownloadOpExec.scala

```diff
@@ -88,32 +89,27 @@ class DownloadOpExec(
   }
 
   def downloadUrl(url: String): String = {
-    var result: String = ""
-    val future = Future {
-      val directory = new File(DOWNLOADS_PATH)
-      if (!directory.exists()) {
-        directory.mkdir()
-      }
-      try {
-        val source = Source.fromURL(url)
-        val data = source.mkString
-        source.close()
-        val filePath = s"$DOWNLOADS_PATH/${UUID.randomUUID()}.txt"
-        val file = new File(filePath)
-        val bw = new BufferedWriter(new FileWriter(file))
-        bw.write(data)
-        bw.close()
-        filePath
-      } catch {
-        case e: Exception => e.getMessage
-      }
-    }
     try {
-      result = Await.result(future, 5.seconds)
+      Await.result(
+        Future {
+          val input = new URL(url).openStream()
+          if (input.available() > 0) {
+            UserFileResource
+              .saveUserFileSafe(
+                uid,
+                s"${UUID.randomUUID()}.download",
```

Comment:
for the file name, we can use the workflow ID and execution ID as prefix, then use the UR as suffix , like`w1-e1-url1.download, w1-e1-url2.download`, the workflow ID and execution ID are also in the context.